src/core/MOM_dynamics_split_RK2.F90

```diff
@@ -1334,6 +1344,8 @@ subroutine initialize_dyn_split_RK2(u, v, h, uh, vh, eta, Time, G, GV, US, param
   CS%id_veffA = register_diag_field('ocean_model', 'veffA', diag%axesCvL, Time, &
        'Effective V-Face Area', 'm^2', conversion = GV%H_to_m*US%L_to_m, &
        x_cell_method='sum', v_extensive = .true.)
+  CS%id_deta_dt = register_diag_field('ocean_model', 'deta_dt', diag%axesT1, Time, &
+      'Barotropic SSH tendency', trim(thickness_units)//' s-1', conversion=GV%H_to_MKS/US%T_to_s)
```

Comment:
I think the plan is to combine this quantity with `id_dynamics_h_tendency` on a zstar grid, to get wstar. The register call for `id_dynamics_h_tendency` has units equal to `'m s-1'` and `conversion=GV%H_to_m*US%s_to_T`. In non-Boussinesq mode, these diagnostics will have different units. Is that problematic?

Reply:
In non-Boussinesq mode, shouldn't the 'h' `tendency have different units as well since the w* coordinate itself won't actually be in meters?

Reply:
Yes, The variable eta in the barotropic solver does take on different units and meaning when run in non-Boussinesq mode, in which case it is column mass anomalies (in kg m-2), not sea-surface height anomalies (in m), If the tendency of this variable is what you are trying to diagnoses the conversion factor would be `conversion=GV%H_to_MKS*US%s_to_T`.  However, if this is the diagnostic that you are trying to add, the long name should also be modified in non-Boussinesq mode to read something like 'Barotropic column-mass tendency'.  The conversion factor that Keith proposes would rescale eta back to height (in m) by multiplying by a Boussinesq-like constant rho_0 when in non-Boussinesq mode.

Reply:
My inclination would be to keep it in mass units in non-Boussinesq cases to be consistent with the horizontal transports (which I believe change units/descriptions as well). Keeping everything in the correct units (i.e. those consistent with the dynamics) seems like it would give us our best shot of closing the thickness budgets on a diagnostic z* (non-Boussinesq) grid. Otherwise, the multiplcation by the `rho_0` will lead to residuals which aren't near roundoff.